### DISCUSSION TITLE
GMT 6.3/6.4 vs GMT 6.0

### DISCUSSION THREAD

**Discussion by chintals** [gmtsar/gmtsar#348] [General]
Hi, 
This thread is about discussing the differences, and performance upgrades when using GMT 6.3 instead of GMT 6.0 (pre_installed with Ubuntu 20.04 LTS)

From using the GMT 6.3 (build from source code), and GMTSAR 6.2.2 on Ubuntu 20.04 LTS (Ubuntu 18 LTS comes pre_installed with GMT 5.4), I have noticed better registration and interferograms for  2-3 bursts scenario from sentinel products. I have also seen much better registration and interferograms for 2 burst cases when using product merge (last burst in SLC1 and first burst in SLC2).

At the moment ESD mode 2 (interpolation) fails during image alignment due to one increment shift in x, and y coordinates when using grdfilter in GMT 6.3. Reverted to using ESD mode 1 (median) works fine.

Am curious if anyone else is using GMT 6.3/6.4, and have noticed better performance or issues?

Note: GMT 6.4 version is in the process of being released soon

**Comment by Xiaohua-Eric-Xu** [comment]
A couple of things were updated recently or relatively recently that may be related to the 2-3 burst scenario or what you experienced. 

1. ext_orb_s1a now gives longer orbit for short bursts scenario so now you have more records to interpolate (for 1 burst especially).
2. I modified the ESD mode 2 to get rid prior error with increment x and y (somehow incompatible with size) and the error was gone on Mac and Ubuntu 18.04, but I did not have a 20.04 to test it. Mode 2 is for bivariate interpolation which mainly targets ionospheric correction, for most cases mode 1 is good to account for azimuthal misregistration.
3. I tuned the interpolation tension with GMT surface -T0.1(seems before 0.5 was not a problem somehow), and now topographic phase is "better" corrected.
4. I haven't noticed any big change between 6.3 and 6.4 on Mac and Ubuntu 18.04. 
5. My suspicion is the rounding on these systems are a bit different?

Any thoughts on making this robust?

**Comment by chintals** [comment]
Hi @Xiaohua-Eric-Xu 
I actually configured my system with GMTSAR 6.2.2 and GMT 6.0.0, which yielded not great results (artifact-fringes in the burst corners), but switching to GMT 6.3 eliminated the artifact-fringes on the burst edges. I initially did think it was updating ext_orb_s1a would solve the 2-burst issues, but it did not with GMT 6.0, but the artifact-fringes disappeared with GMT 6.3.

**Reply by Xiaohua-Eric-Xu** [reply]
Do you have a plot for what you described? 

If I remember correctly, there was a parameter change in gmt on x_inc and y_inc or x_column and y_columns, likely from GMT 6.2 to 6.3 if I remember correctly. It's better to use newer version always.  But this has nothing to do with the fringes you described I assume.

**Comment by chintals** [comment]
Hi @Xiaohua-Eric-Xu The figures below is shows interferograms (3 burst on left - F1, 4 on the right F2)  Figure A is ESD mode 2, and Figure B is ESD mode 1. I had more examples for 2 burst scenario with and without product merge  for GMT 6.0, and 6.3 (unfrotunately had to delete to recoup harddisk space); and GMT 6.3 seemed to show slightly better interferograms. I assumed ESD mode 2 would not be good for a 2-burst case, but was surprised to see such difference in 3-4 burst case shown below.
![A](https://user-images.githubusercontent.com/25990795/174634239-59c1df12-4785-418f-aa86-2de44625a4c7.jpg)
![B](https://user-images.githubusercontent.com/25990795/174634247-ba1f303a-4798-42ca-9a74-d2c5f3859fcd.jpg)

**Reply by Xiaohua-Eric-Xu** [reply]
I think mode 2 is only used when you want to correct ionosphere. Otherwise 1 is more robust. Mode 2 is a map-wise green-spline fill that's not very robust when you only give it 1 or 2 strips (burst overlap) to interpolate. The fringes are likely interpolation error. 

I initially wrote mode 2 to be the more robust surface+filtering approach, but it was too slow.

**Comment by chintals** [comment]
@Xiaohua-Eric-Xu Thank you for the explanation. ESD mode 2 clear to me now.

**Comment by chintals** [comment]
> If I remember correctly, there was a parameter change in gmt on x_inc and y_inc or x_column and y_columns, likely from GMT 6.2 to 6.3 if I remember correctly. It's better to use newer version always..

yes, we can test the correct usage for 6.3. The error with x_column and y_column is happening at this step.
`gmt grdfilter test.grd -D0 -Fg8000/1500 -Gtest2.grd -V `
line 228 in preproc_batch_tops_esd.csh
